private/buf/bufgen/migrate.go

```diff
@@ -0,0 +1,553 @@
+		}
+		for module, override := range objcClassPrefixConfig.Override {
+			moduleOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+				FileOption: bufgenv2.FileOptionObjcClassPrefix.String(),
+				Module:     module.IdentityString(),
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, moduleOverrideRule)
+		}
+	}
+	if rubyPackageConfig := managedConfigV1.RubyPackageConfig; rubyPackageConfig != nil {
+		for _, excludedModule := range rubyPackageConfig.Except {
+			moduleDisableRule := bufgenv2.ExternalManagedDisableConfigV2{
+				FileOption: bufgenv2.FileOptionRubyPackage.String(),
+				Module:     excludedModule.IdentityString(),
+			}
+			managedConfigV2.Disable = append(managedConfigV2.Disable, moduleDisableRule)
+		}
+		for module, override := range rubyPackageConfig.Override {
+			moduleOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+				FileOption: bufgenv2.FileOptionRubyPackage.String(),
+				Module:     module.IdentityString(),
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, moduleOverrideRule)
+		}
+	}
+	for fileOption, fileToOverride := range managedConfigV1.Override {
+		for file, override := range fileToOverride {
+			fileOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+				FileOption: strings.ToLower(fileOption),
+				Path:       file,
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, fileOverrideRule)
+		}
+	}
+	return &managedConfigV2
+}
+
+func getExternalInputConfigV2(
+	ctx context.Context,
+	logger *zap.Logger,
+	input string,
+	types []string,
+	includePaths []string,
+	excludedPaths []string,
+) (*bufgenv2.ExternalInputConfigV2, error) {
+	inputConfig := bufgenv2.ExternalInputConfigV2{}
+	path, options, err := bufref.GetRawPathAndOptions(input)
+	if err != nil {
+		return nil, err
+	}
+	format, err := buffetch.NewRefParser(logger).GetRefFormat(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+	switch format {
+	case buffetch.FormatBin:
+		inputConfig.BinaryImage = &path
+	case buffetch.FormatBingz:
+		inputConfig.BinaryImage = &path
+		compression := "gzip"
+		inputConfig.Compression = &compression
+	case buffetch.FormatDir:
+		inputConfig.Directory = &path
+	case buffetch.FormatGit:
+		inputConfig.GitRepo = &path
+	case buffetch.FormatJSON:
+		inputConfig.JSONImage = &path
+	case buffetch.FormatJSONGZ:
+		inputConfig.JSONImage = &path
+		compression := "gzip"
+		inputConfig.Compression = &compression
+	case buffetch.FormatMod:
+		inputConfig.Module = &path
+	case buffetch.FormatTar:
+		inputConfig.Tarball = &path
+	case buffetch.FormatTargz:
+		inputConfig.Tarball = &path
+		compression := "gzip"
+		inputConfig.Compression = &compression
+	case buffetch.FormatZip:
+		inputConfig.ZipArchive = &path
+	case buffetch.FormatProtoFile:
+		inputConfig.ProtoFile = &path
+	// TODO: txtpb and jsonpb
```

Comment:
Should be in this PR optimally

Reply:
`Txtpb` and `binpb` (not `jsonpb`, the comment was wrong) are now implemented in this branch as well